src/client/Microsoft.Identity.Client/Internal/Requests/ClientCredentialRequest.cs

```diff
@@ -42,6 +42,12 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
 
             AuthenticationResult authResult = null;
 
+            if(AuthenticationRequestParameters.Authority.TenantId.Contains("common") ||
```

Comment:
you'll get a null ref for ADFS where tenantId is null. Also, the comparison should be done in a case insensitive manner. Have a look at `AadAuthority.IsCommonOrganizationsOrConsumersTenant`

Reply:
I was looking for something like AadAuthority.IsCommonOrganizationsOrConsumersTenant
I will update